src/sentry/static/sentry/app/views/settings/projectIncidentRules/ruleForm.tsx

```diff
@@ -0,0 +1,193 @@
+import {debounce} from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {EventsStatsData} from 'app/types';
+import {PanelAlert} from 'app/components/panels';
+import {t} from 'app/locale';
+import Form from 'app/views/settings/components/forms/form';
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+
+import IncidentRulesChart from './chart';
+import {AlertRuleAggregations, AlertRuleThresholdType} from './constants';
+
+type Props = {
+  data: EventsStatsData;
+};
+
+type State = {
+  width?: number;
+  upperBound: number;
+};
+
+const START_POSITION = 20;
+
+class RuleForm extends React.Component<Props, State> {
+  static contextTypes = {
+    form: PropTypes.any,
+  };
+
+  static defaultProps = {
+    data: [],
+  };
+
+  state = {
+    upperBound: START_POSITION,
+  };
+
+  handleChangeUpperBoundInput = debounce(upperBound => {
+    this.setState({upperBound});
+  }, 50);
+
+  handleChangeUpperBound = upperBound => {
+    this.setState({upperBound});
+    this.context.form.setValue('alertThreshold', Math.round(upperBound));
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <IncidentRulesChart
+          onChangeUpperBound={this.handleChangeUpperBound}
+          upperBound={this.state.upperBound}
+          data={[
+            {
+              seriesName: 'Test',
```

Comment:
Is this supposed to be here still?

Reply:
Kind of, it'll be removed in a follow up PR.

Reply:
https://github.com/getsentry/sentry/pull/14271/files#diff-e6ab9feacb364c80111abe98d035e667L53